File: Top 50 Coding Interview Questions/23. Write a Program to Find the Sum of Natural Numbers using Recursion/sumofnaturalnumbers.py
# Function to find the sum of natural numbers using recursion
def sum_of_natural_numbers(n):
    if n == 0:
        return 0  # Base case: The sum of numbers up to 0 is 0
    return n + sum_of_natural_numbers(n - 1)  # Recursive call

# Input from the user
num = int(input("Enter a positive integer: "))

if num < 0:
    print("Please enter a positive integer.")
else:
    print(f"The sum of the first {num} natural numbers is: {sum_of_natural_numbers(num)}")

File: Top 50 Coding Interview Questions/23. Write a Program to Find the Sum of Natural Numbers using Recursion/SumofNaturalNumbers.cpp
#include <iostream>
using namespace std;

// Function to find the sum of natural numbers using recursion
int sumOfNaturalNumbers(int n) {
    if (n == 0) {
        return 0; // Base case: The sum of numbers up to 0 is 0
    }
    return n + sumOfNaturalNumbers(n - 1); // Recursive call
}

int main() {
    int num;
    cout << "Enter a positive integer: ";
    cin >> num;

    if (num < 0) {
        cout << "Please enter a positive integer." << endl;
    } else {
        cout << "The sum of the first " << num << " natural numbers is: " 
             << sumOfNaturalNumbers(num) << endl;
    }

    return 0;
}
